### PR TITLE
Fix for query* functions and tests

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Command/Command.php [hunk 1]
```diff
@@ -467,30 +467,40 @@ public function query(): DataReader
...
-     * @psalm-suppress MixedReturnStatement
-     * @psalm-suppress MixedInferredReturnType
-     */
     public function queryAll(): array
     {
-        return $this->queryInternal((int) static::QUERY_MODE_ALL);
+        /** @psalm-var array<array-key, array<mixed>>|null */
+        $results = $this->queryInternal((int) static::QUERY_MODE_ALL);
+
+        if (is_array($results)) {
```
Comment (terabytesoftw) [inline src/Command/Command.php:475]:
```suggestion
    return match (is_array($results) {
        true => $results,
        false => [],
    };
```

File: src/Command/Command.php [hunk 2]
```diff
@@ -467,30 +467,40 @@ public function query(): DataReader
...
     }
 
-    public function queryColumn(): array
+    public function queryColumn(): array|false
     {
         /** @psalm-var array<array-key, array<mixed>>|null */
         $results = $this->queryInternal((int) static::QUERY_MODE_COLUMN);
 
-        if (is_array($results)) {
+        if (is_array($results) && count($results)) {
```
Comment (terabytesoftw) [inline src/Command/Command.php:487]:
```suggestion
    return match (is_array($results) && count($results)) {
        true => $results,
        false => false,
    };
```

File: src/Command/Command.php [hunk 3]
```diff
@@ -467,30 +467,40 @@ public function query(): DataReader
...
     }
 
-    public function queryOne(): mixed
+    public function queryOne(): array|false
     {
-        return $this->queryInternal((int) static::QUERY_MODE_ROW);
+        /** @psalm-var array<array-key, array<mixed>>|null */
+        $results = $this->queryInternal((int) static::QUERY_MODE_ROW);
+
+        if (is_array($results) && count($results)) {
```
Comment (terabytesoftw) [inline src/Command/Command.php:499]:
```suggestion
    return match (is_array($results) && count($results)) {
        true => $results,
        false => false,
    };
```

File: src/Query/Query.php [hunk 4]
```diff
@@ -285,7 +285,11 @@ public function column(): array
         }
 
         if ($this->indexBy === null) {
-            return $this->createCommand()->queryColumn();
+            $result = $this->createCommand()->queryColumn();
+            if (is_array($result)) {
```
Comment (terabytesoftw) [inline src/Query/Query.php:289]:
```suggestion
    return match (is_array($result)) {
        true => $result,
        false => [],
    };
```

**Review by terabytesoftw — CHANGES_REQUESTED** [review]

File: src/Command/Command.php (hunk 1 above)
Comment (darkdef) [reply src/Command/Command.php:475]: It's not so readable

**Review by terabytesoftw — APPROVED** [review]

**Pull request merged**